File: Data Structures & Algorithms/house-robber/submission-1.java
class Solution {
    public int rob(int[] nums) {
        int n = nums.length;
        //prevNo 不抢前一个房子
        //prevYes 抢前一个房子
        int prevNo = 0;
        int prevYes = nums[0];
        for (int i = 1; i < n; i ++) {
            int curNo = Math.max(prevNo, prevYes);
            int curYes = prevNo + nums[i];
            prevNo = curNo;
            prevYes = curYes;
        }
        return Math.max(prevNo, prevYes);
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = {}
        for i in range(len(nums)):
            b = target - nums[i]
            if target - nums[i] in m:
                return [m[b], i]
            m[nums[i]] = i
        return []